### PR TITLE
Racheli

### PR DESCRIPTION
**Pull request by dkracheli**

### CODE REVIEW

**Comment by shaharariel95** [comment]
There is UI for non working stuff - not good
There are some code duplication, we need to solve it
there are some free string that need to be in another file
Great Job with the working POC and the UI, and the JWT !

File: controllers/loginController.js [hunk 1]
```diff
@@ -1,24 +1,23 @@
-const {userExist, statusCheck, validPassword, getToken} = require("../services/authService");
+const {userExist, statusCheck, validPassword, generateAccessToken} = require("../services/authService");
 const {generatePassword, sendEmailPassword} = require("../services/forgotPassword");
 
 const {User} = require("../services/authService");
 const bcrypt = require("bcrypt");
-const jwt = require("jsonwebtoken");
+const axios = require('axios')
```
Comment (DaveAvigad) [inline controllers/loginController.js:6]: missing semi-colon
still not using ESlint!

File: index.js [hunk 2]
```diff
@@ -3,22 +3,24 @@ require("dotenv").config({path: '.env'});
...
+app.use(cookieParser());
 app.use(bodyParser.json());
 app.use(express.json());
 app.use(cors());
 
-
 app.use("/login", login.loginRouter);
 app.use("/register", register.registerRoute);
-app.use("/homePage", isAuthenticated, homePage.homePageRouter);
+app.use("/homePage", authenticateToken, homePage.homePageRouter);
```
Comment (DaveAvigad) [inline index.js:23]: home page is a view. its not related to the api by any means

File: middleware/authenticate.js [hunk 3]
```diff
@@ -1,28 +1,32 @@
...
+// // authJwt.post('/refreshToken', async (req, res) => {
+// //     const refreshToken = req.body.refreshToken;
+// //     if (refreshToken == null) return res.sendStatus(401)
+// //     // const user = await User.retrieve({refreshToken: refreshToken})
+// //     // if (!user) return res.sendStatus(403)
+// //     jwt.verify(refreshToken, process.env.REFRESH_TOKEN_SECRET, (err, user) => {
+// //         if (err) return res.sendStatus(403)
+// //         const accessToken = generateAccessToken({email: user.email})
+// //         //  res.setHeader("authorization", "Bearer" + " " + accessToken)
+// //         req.token = {accessToken: accessToken};
```
Comment (DaveAvigad) [inline middleware/authenticate.js:16]: do not commit commented-out code.

BTW I don't see any review of the other teammates.

File: middleware/validator.js [hunk 4]
```diff
@@ -1,4 +1,67 @@
...
+            expiresIn: '5s'
+        });
+
+        const refreshToken = jwt.sign({
+            email: req.body.email
+        }, process.env.REFRESH_TOKEN_SECRET, {expiresIn: '1d'});
+        res.cookie('jwt', refreshToken, {
+            httpOnly: true,
+            sameSite: 'None', secure: true,
+            maxAge: 24 * 60 * 60 * 1000
```
Comment (DaveAvigad) [inline middleware/validator.js:19]: `24 * 60 * 60 * 1000` move this into constants/ config file

File: middleware/validator.js [hunk 5]
```diff
@@ -1,4 +1,67 @@
...
+        req.token = {refreshToken: refreshToken, accessToken: "Bearer" + " " + accessToken}
+        next()
+    } catch (e) {
+        return res.sendStatus(401);
+    }
+}
+
+async function authenticateToken(req, res, next) {
+    const authHeader = req.headers['authorization']
+    let token = authHeader && authHeader.split(' ')[1]
```
Comment (DaveAvigad) [inline middleware/validator.js:32]: you can use es6 optional chaining feature here instead

**Review by DaveAvigad — COMMENTED** [review]
left some comments.
please use eslint
please ask the team to review and approve your code as well

**Pull request merged**